cqrs-user-query-service/src/main/java/dev/amir/userquery/application/usecase/UserUseCases.java

```diff
@@ -4,22 +4,47 @@
 import dev.amir.userquery.application.port.output.UserOutputPort;
 import dev.amir.userquery.domain.entity.User;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Optional;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+@Slf4j
 @RequiredArgsConstructor
 @Service
 public class UserUseCases implements UserInputPort {
     private final UserOutputPort userOutputPort;
 
     @Override
     public Collection<User> getAllUsers() {
-        return userOutputPort.getAll();
+        log.info("Getting all users");
+        try {
+            Collection<User> users = userOutputPort.getAll();
+            log.info("Users quantity: {}", users.size());
+            return users;
+        } catch (Exception e) {
+            log.error("Error while getting all users", e);
+            return Collections.emptyList();
+        }
     }
 
     @Override
     public Optional<User> getUserById(String userId) {
-        return userOutputPort.getById(userId);
+        log.info("Getting user");
+        try {
+            Optional<User> user = userOutputPort.getById(userId);
+
+            if (user.isPresent()) {
```

Comment:
instead of this if statement just for log you can change the message and reduce if-else to only one line:
`log.info("User with ID {} was found: {}", userId, user.isPresent());`